sycl/source/detail/graph_impl.cpp

```diff
@@ -1024,7 +1024,7 @@ exec_graph_impl::enqueue(const std::shared_ptr<sycl::detail::queue_impl> &Queue,
       for (std::vector<sycl::detail::EventImplPtr>::iterator It =
                MExecutionEvents.begin();
            It != MExecutionEvents.end();) {
-        auto Event = *It;
+        std::shared_ptr<event_impl> &Event = *It;
```

Comment:
```suggestion
        auto &Event = *It;
```
This should work too if we want to have it less explicit as before and only use a reference ... or was that change intentional?

Reply:
The change was intentional. I thought having it explicit was slightly more readable. But I don't feel that strongly about it. I'm ok with reverting if you have a preference for auto.

Reply:
I'm okay making it explicit, but you also might want consider changing use of `auto &` in line 1029 for consistency.